src/components/pipeline/EditCreatePipeline.js

```diff
@@ -0,0 +1,270 @@
+import { withRouter } from 'react-router';
+import PropTypes from 'prop-types';
+
+import { withStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import Grid from '@material-ui/core/Grid';
+import Container from '@material-ui/core/Container';
+import TextField from '@material-ui/core/TextField';
+import SaveIcon from '@material-ui/icons/Save';
+import Button from '@material-ui/core/Button';
+import Accordion from '@material-ui/core/Accordion';
+import AccordionSummary from '@material-ui/core/AccordionSummary';
+import AccordionDetails from '@material-ui/core/AccordionDetails';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import CodeIcon from '@material-ui/icons/Code';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import AddIcon from '@material-ui/icons/Add';
+import Divider from '@material-ui/core/Divider';
+import AccordionActions from '@material-ui/core/AccordionActions';
+
+import Main from '../common/Main';
+import BackButton from '../common/BackButton';
+
+import {
+  EDIT_PIPELINE_NAME_ID,
+  EDIT_PIPELINE_DESCRIPTION_ID,
+  EDIT_PIPELINE_SAVE_BUTTON_ID,
+  EDIT_PIPELINE_BACK_BUTTON_ID,
+} from '../../config/selectors';
+
+const styles = (theme) => ({
+  backButton: {
+    position: 'fixed',
+    right: theme.spacing(2),
+    bottom: theme.spacing(2),
+  },
+  pipelineEdit: {
+    width: '100%',
+  },
+  heading: {
+    fontSize: theme.typography.pxToRem(15),
+    fontWeight: theme.typography.fontWeightRegular,
+  },
+  addAlgorithmButton: {
+    fontSize: theme.typography.pxToRem(15),
+    fontWeight: theme.typography.fontWeightRegular,
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  content: {
+    padding: theme.spacing(2),
+    display: 'flex',
+    flexDirection: 'column',
+  },
+});
+
+const mockPipeline = {
+  pipelines: [
+    {
+      id: 'pipeline1',
+      name:
+        'Pipel. from hash-users → sanitize-users → suppress-fields → shuffle-fields',
+      algorithms: [
+        { id: 'hash-users', name: 'Hash users' },
+        { id: 'sanitize-users', name: 'Sanitize users' },
+        { id: 'suppress-fields', name: 'Suppress fields' },
+        { id: 'shuffle-fields', name: 'Shuffle fields' },
+      ],
+    },
+    {
+      id: 'pipeline2',
+      name: 'Pipel. from hash-users → k-anonymize-geolocation → shuffle-fields',
+      algorithms: [
+        { id: 'hash-users', name: 'Hash users' },
+        { id: 'k-anonymize-geolocation', name: 'k-Anonymize geolocation' },
+        { id: 'shuffle-fields', name: 'Shuffle fields' },
+      ],
+    },
+  ],
+};
+
+const EditCreatePipeline = (props) => {
+  const { t, classes } = props;
+
+  const [expanded, setExpanded] = useState(false);
+  const [nameP, setNameP] = useState('');
+  const [descriptionP, setDescriptionP] = useState('');
+  const [pipeline, setPipeline] = useState(
+    mockPipeline.pipelines[0].algorithms,
+  );
```

Comment:
this one is perhaps `pipelineAlgorithms`, since that's the key you're accessing in the mock data